shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1232,25 +1231,18 @@ - (NSString*)textInRange:(UITextRange*)range {
 // Replace the text within the specified range with the given text,
 // without notifying the framework.
 - (void)replaceRangeLocal:(NSRange)range withText:(NSString*)text {
-  NSRange selectedRange = _selectedTextRange.range;
-
-  // Adjust the text selection:
-  // * reduce the length by the intersection length
-  // * adjust the location by newLength - oldLength + intersectionLength
-  NSRange intersectionRange = NSIntersectionRange(range, selectedRange);
-  if (range.location <= selectedRange.location) {
-    selectedRange.location += text.length - range.length;
-  }
-  if (intersectionRange.location != NSNotFound) {
-    selectedRange.location += intersectionRange.length;
-    selectedRange.length -= intersectionRange.length;
-  }
-
   [self.text replaceCharactersInRange:[self clampSelection:range forText:self.text]
                            withString:text];
-  [self setSelectedTextRangeLocal:[FlutterTextRange
-                                      rangeWithNSRange:[self clampSelection:selectedRange
-                                                                    forText:self.text]]];
+
+  // Adjust the selected range and the marked text range. There's no
+  // documentation but UITextField always sets markedTextRange to nil,
+  // and collapses the selection to the end of the new replacement text.
+  range.location += text.length;
+  range.length = 0;
```

Comment:
Why is this stomping the parameter instead of making a new range (and then replacing that new range with the clamped version)? Writing to parameters is generally bad for readability.